packages/lesswrong/components/ultraFeed/UltraFeedSettings.tsx

```diff
@@ -241,138 +346,196 @@ const UltraFeedSettings = ({
+    const strValue = String(value).trim();
+    const processedValue = strValue === '' ? '' : (isNaN(parseFloat(strValue)) ? '' : parseFloat(strValue));
+    updateForm('threadInterestModel', prevModel => ({ ...prevModel, [field]: processedValue, }));
+  }, [updateForm]);
 
-    settingsToUpdate.quickTakeBoost = isNaN(formValues.quickTakeBoost) 
-        ? DEFAULT_SETTINGS.quickTakeBoost 
-        : formValues.quickTakeBoost;
-        
-    settingsToUpdate.ultraFeedSeenPenalty = formValues.ultraFeedSeenPenalty === '' || isNaN(Number(formValues.ultraFeedSeenPenalty))
-        ? DEFAULT_SETTINGS.ultraFeedSeenPenalty
-        : Number(formValues.ultraFeedSeenPenalty);
+  const handleExploreBiasChange = useCallback((newExploreBiasValue: number) => {
+    setZodErrors(null); 
+    const newLogImpactFactor = 2 - newExploreBiasValue;
+    const newCommentSubscribedAuthorMultiplier = (2 - newExploreBiasValue) * 2.5;
 
-    settingsToUpdate.incognitoMode = formValues.incognitoMode;
-    settingsToUpdate.postTitlesAreModals = formValues.postTitlesAreModals;
+    const totalSourceWeight = Object.values(formValues.sourceWeights).reduce((acc: number, value: number | '') => acc + (value === '' ? 0 : value), 0) || 0;
+    const newSubscribedAuthorSourceWeight = Math.max(1, Math.min(totalSourceWeight * ((0.05 + ((2-newExploreBiasValue) / 2)) * 0.25), totalSourceWeight));
 
+    setFormValues(prev => ({
+      ...prev,
+      threadInterestModel: {
+        ...prev.threadInterestModel,
+        logImpactFactor: newLogImpactFactor,
+      },
+      commentScoring: {
+        ...prev.commentScoring,
+        commentSubscribedAuthorMultiplier: newCommentSubscribedAuthorMultiplier,
+      },
+      sourceWeights: {
+        ...prev.sourceWeights,
+        subscriptions: Math.round(newSubscribedAuthorSourceWeight),
+      }
+      
+    }));
+  }, [setFormValues, formValues.sourceWeights]);
+
+  const handleSave = useCallback(() => {
+    const defaultResolverSettings = DEFAULT_SETTINGS.resolverSettings;
+    const defaultThreadInterestModel = defaultResolverSettings.threadInterestModel;
+    const defaultCommentScoring = defaultResolverSettings.commentScoring;
+
+    const settingsToUpdate: Partial<UltraFeedSettingsType> = {
+      resolverSettings: {
+        sourceWeights: mergeWith(
+          cloneDeep(DEFAULT_SOURCE_WEIGHTS), 
+          formValues.sourceWeights,
+          (defaultWeightVal, formWeightVal) => parseNumericInputAsZeroOrNumber(formWeightVal, 0)
+        ),
+        incognitoMode: formValues.incognitoMode,
+        commentScoring: mergeWith(
+          cloneDeep(defaultCommentScoring),
+          formValues.commentScoring,
+          (defaultVal, formVal, key) => {
+            if (key === 'threadScoreAggregation') {
+              return formVal || defaultVal; // Handles empty string, null, undefined from formVal
+            }
+            return parseNumericInputAsZeroOrNumber(formVal, defaultVal);
+          }
```

Comment:
I assume this is deliberately different from what `deriveFormValuesFromSettings` is doing for `commentScoring`?

Reply:
(relatedly, I don't see that this save operation is doing any of the coalescing to default values that `deriveFormValuesFromSettings` does, either)

Reply:
yes, saving involves more interpretation of the form state that gets stored, deriveFormValues is just hydrating/populating the form